docs/sources/reference/components/otelcol.exporter.kafka.md

```diff
@@ -0,0 +1,323 @@
+
+### sasl block
+
+The `sasl` block configures SASL authentication against Kafka brokers.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`username` | `string` | Username to use for SASL authentication. | | yes
+`password` | `secret` | Password to use for SASL authentication. | | yes
+`mechanism` | `string` | SASL mechanism to use when authenticating. | | yes
+`version` | `number` | Version of the SASL Protocol to use when authenticating. | `0` | no
+
+The `mechanism` argument can be set to one of the following strings:
+
+* `"PLAIN"`
+* `"AWS_MSK_IAM"`
+* `"SCRAM-SHA-256"`
+* `"SCRAM-SHA-512"`
+
+When `mechanism` is set to `"AWS_MSK_IAM"`, the [`aws_msk` child block][aws_msk] must also be provided.
+
+The `version` argument can be set to either `0` or `1`.
+
+### aws_msk block
+
+The `aws_msk` block configures extra parameters for SASL authentication when
+using the `AWS_MSK_IAM` mechanism.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`region` | `string` | AWS region the MSK cluster is based in. | | yes
+`broker_addr` | `string` | MSK address to connect to for authentication. | | yes
+
+### tls block
+
+The `tls` block configures TLS settings used for connecting to the Kafka
+brokers. If the `tls` block isn't provided, TLS won't be used for
+communication.
+
+{{< docs/shared lookup="reference/components/otelcol-tls-client-block.md" source="alloy" version="<ALLOY_VERSION>" >}}
+
+### kerberos block
+
+The `kerberos` block configures Kerberos authentication against the Kafka
+broker.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`service_name` | `string` | Kerberos service name. | | no
+`realm` | `string` | Kerberos realm. | | no
+`use_keytab` | `string` | Enables using keytab instead of password. | | no
+`username` | `string` | Kerberos username to authenticate as. | | yes
+`password` | `secret` | Kerberos password to authenticate with. | | no
+`config_file` | `string` | Path to Kerberos location (for example, `/etc/krb5.conf`). | | no
+`keytab_file` | `string` | Path to keytab file (for example, `/etc/security/kafka.keytab`). | | no
+
+When `use_keytab` is `false`, the `password` argument is required. When
+`use_keytab` is `true`, the file pointed to by the `keytab_file` argument is
+used for authentication instead. At most one of `password` or `keytab_file`
+must be provided.
+
+### metadata block
+
+The `metadata` block configures how to retrieve and store metadata from the
+Kafka broker.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`include_all_topics` | `bool` | When true, maintains metadata for all topics. | `true` | no
```

Comment:
this is called `full` in the otel. Should we have the same name to avoid confusions?

Reply:
It may have been a deliberate decision to make the name more explicit in Alloy. Since this is also what `otelcol.receiver.kafka` uses, I'll leave it as it is for now.